discord/automod.py

```diff
@@ -81,33 +84,43 @@ class AutoModRuleAction:
 
     __slots__ = ('type', 'channel_id', 'duration', 'custom_message')
 
+    @overload
+    def __init__(self, *, channel_id: Optional[int] = ...) -> None:
+        ...
+
+    @overload
+    def __init__(self, *, duration: Optional[datetime.timedelta] = ...) -> None:
+        ...
+
+    @overload
+    def __init__(self, *, custom_message: Optional[str] = ...) -> None:
+        ...
+
     def __init__(
         self,
         *,
         channel_id: Optional[int] = None,
         duration: Optional[datetime.timedelta] = None,
         custom_message: Optional[str] = None,
     ) -> None:
+        self.type: AutoModRuleActionType = AutoModRuleActionType.block_message
+        if channel_id:
+            self.type = AutoModRuleActionType.send_alert_message
+        elif duration:
+            self.type = AutoModRuleActionType.timeout
+
```

Comment:
What's the point of this refactoring?

Reply:
Well, actually there's no particular reason.